### PR TITLE
[SPARK-47023][BUILD] Upgrade `aircompressor` to 0.26

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade `aircompressor` to 0.26.

### Why are the changes needed?

`aircompressor` v0.26 has the following bug fixes.
 
- [Fix out of bounds read/write in Snappy decompressor](https://github.com/airlift/aircompressor/commit/b89db180bb97debe025b640dc40ed43816e8c7d2)
- [Fix ZstdOutputStream corruption on double close](https://github.com/airlift/aircompressor/commit/b89db180bb97debe025b640dc40ed43816e8c7d2)

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2616,6 +2616,11 @@
           </exclusion>
         </exclusions>
       </dependency>
+      <dependency>
+        <groupId>io.airlift</groupId>
+        <artifactId>aircompressor</artifactId>
```
Comment (sunchao) [inline pom.xml:2621]: just curious why we need to bring a new dependency for this: where is the old dependency (`aircompressor` 0.25) come from?

Comment (dongjoon-hyun) [reply pom.xml:2621]: Basically, it came from Apache ORC library.

Comment (dongjoon-hyun) [reply pom.xml:2621]: Here
- https://mvnrepository.com/artifact/org.apache.orc/orc-core/1.9.2

Comment (dongjoon-hyun) [reply pom.xml:2621]: Also, Parquet also has some dependency.
- https://mvnrepository.com/artifact/org.apache.parquet/parquet-hadoop/1.13.1

However, I'm not sure about Parquet part. This PR aims to override it from Apache Spark side.

**Review by sunchao — APPROVED** [review]
LGTM

File: pom.xml (hunk 1 above)
Comment (sunchao) [reply pom.xml:2621]: I see, thanks

**Comment by dongjoon-hyun** [comment]
Thank you, @sunchao !

If new libraries come in the future, we will remove this override back.

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Since the RC1 vote fails, I backported this to branch-3.5.

**Pull request closed without merging**